openedx/features/custom_fields/multiselect_with_other/db/fields.py

```diff
@@ -93,8 +93,9 @@ def to_python(self, value):
             if isinstance(value, list):
                 return value
             elif isinstance(value, string_type):
-                choices_str = value.replace(self.other_delimiter, '')
-                selected_choices = [choice for choice in choices_str.split(',') if choice.strip()]
+                choices_str = value.split(self.other_delimiter)
+                selected_choices = [choice for choice in choices_str[0].split(',') if choice.strip()]
```

Comment:
It would be good idea to put a length check before using `.split()` here

Reply:
`split()` works with empty string. It won't crash, so if we are passing empty string it return empty string. Does that cater your concern?

Reply:
do we have any chance of of `value=[None, '']`? if not then it looks good.

Reply:
We will always have something or empty string(''), It won't be None in any case and 'split()' works with empty string('').

Reply:
if choice.strip() or if choices_str[0] ? We should also save choices_str[0] in a meaning variable.

Reply:
It was previous code written by @qasimgulzar I had him checked, You second point is done, I changed its name.